File: backend/stockmate-backend/stockmate-backend/src/main/java/com/stockmate/stockmate_backend/dto/ProductResponse.java
package com.stockmate.stockmate_backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.stockmate.stockmate_backend.entity.ProductCategory;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class ProductResponse {
    private Long productId;
    private String name;
    private ProductCategory category;
    private String size;
    private BigDecimal price;
    private Integer quantity;
    private Integer lowStockThreshold;

    @JsonProperty("isLowStock")
    private boolean isLowStock;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: android/app/src/main/java/edu/cit/cortes/stockmate/model/ProductResponse.kt
package edu.cit.cortes.stockmate.model

import com.google.gson.annotations.SerializedName

data class ProductResponse(
    @SerializedName("productId") val productId: Long,
    @SerializedName("name") val name: String,
    @SerializedName("category") val category: String,
    @SerializedName("size") val size: String?,
    @SerializedName("price") val price: Double,
    @SerializedName("quantity") val quantity: Int,
    @SerializedName("lowStockThreshold") val lowStockThreshold: Int,
    @SerializedName("isLowStock") val isLowStock: Boolean,
    @SerializedName("createdAt") val createdAt: String? = null,
    @SerializedName("updatedAt") val updatedAt: String? = null
)
